File: src/dataset/go/roman_numerals_Exercism_go_drapala/openAI/ChatGPT4_roman_numerals_v2.go
package romannumerals

import "fmt"

var romanMappings = [][]string{
	{"", "I", "II", "III", "IV", "V", "VI", "VII", "VIII", "IX"},
	{"", "X", "XX", "XXX", "XL", "L", "LX", "LXX", "LXXX", "XC"},
	{"", "C", "CC", "CCC", "CD", "D", "DC", "DCC", "DCCC", "CM"},
	{"", "M", "MM", "MMM"},
}

func ToRomanNumeral(input int) (string, error) {
	if input < 1 || input > 3000 {
		return "", fmt.Errorf("number must be between 1 and 3000")
	}

	thousands := romanMappings[3][input/1000]
	hundreds := romanMappings[2][(input/100)%10]
	tens := romanMappings[1][(input/10)%10]
	ones := romanMappings[0][input%10]

	return thousands + hundreds + tens + ones, nil
}

File: src/dataset/cpp/roman_numerals_Exercism_johnngugi/openAI/ChatGPT4_roman_numerals_v2.cpp
#include "roman_numerals.h"
#include <array>
#include <string>

namespace roman
{
    const std::array<const char*, 10> ones = {"", "I", "II", "III", "IV", "V", "VI", "VII", "VIII", "IX"};
    const std::array<const char*, 10> tens = {"", "X", "XX", "XXX", "XL", "L", "LX", "LXX", "LXXX", "XC"};
    const std::array<const char*, 10> hundreds = {"", "C", "CC", "CCC", "CD", "D", "DC", "DCC", "DCCC", "CM"};
    const std::array<const char*, 4> thousands = {"", "M", "MM", "MMM"};

    std::string convert(int n)
    {
        return std::string(thousands[n / 1000]) +
               hundreds[(n / 100) % 10] +
               tens[(n / 10) % 10] +
               ones[n % 10];
    }
}